esmf_regrid/experimental/unstructured_scheme.py

```diff
@@ -10,6 +11,62 @@
 from esmf_regrid.experimental.unstructured_regrid import MeshInfo
 
 
+def _map_complete_blocks(src, func, dims, out_sizes):
+    """
+    Apply a function to complete blocks.
+
+    Complete means that the data is not chunked along the chosen dimensions.
+
+    Parameters
+    ----------
+    src : cube
+        Source :class:`~iris.cube.Cube` that function is applied to.
+    func : function
+        Function to apply.
+    dims : tuple of int
+        Dimensions that cannot be chunked.
+    out_sizes : tuple of int
+        Output size of dimensions that cannot be chunked.
+
+    Returns
+    -------
+    cube
+        A new iris.cube.Cube instance.
+
+    """
+    if not src.has_lazy_data():
+        return func(src.data)
+
+    data = src.lazy_data()
+
+    # Ensure dims are not chunked
+    in_chunks = list(data.chunks)
+    for dim in dims:
+        in_chunks[dim] = src.shape[dim]
+    data = data.rechunk(in_chunks)
```

Comment:
is this a no-op if the loop above doesn't occur?

Reply:
I suppose it must be, at least effectively. Also if the chunks already span the horizontal dimensions. It's worth bearing in mind that this is existing iris behaviour https://github.com/SciTools/iris/blob/d3b4c41b245ef6b4eb681f704469e991db149ba1/lib/iris/_lazy_data.py#L381